src/stores/contactmoment/index.ts

```diff
@@ -1,102 +1,151 @@
 
-interface ContactmomentState {
-  contactmomentLoopt: boolean;
+export interface Vraag {
   zaken: ContactmomentZaak[];
-  klanten: { klant: Klant; shouldStore: boolean }[];
   notitie: string;
-  contactverzoek: { url: string; medewerker: string } | undefined;
-  nieuweKlant: NieuweKlant | undefined;
+  contactverzoek: ContactmomentContactVerzoek;
   startdatum: string;
+  kanaal: string;
+  resultaat: string;
+  klanten: { klant: Klant; shouldStore: boolean }[];
+}
+
+function initVraag(): Vraag {
+  return {
+    zaken: [],
+    notitie: "",
+    contactverzoek: {
+      url: "",
+      attendee: "",
+      isDirty: false,
+      isSubmitted: false,
+    },
+    startdatum: getFormattedUtcDate(),
+    kanaal: "",
+    resultaat: "",
+    klanten: [],
+  };
+}
+
+interface ContactmomentState {
+  contactmomentLoopt: boolean;
+  vragen: Vraag[];
+  huidigeVraag: Vraag;
 }
 
 export const useContactmomentStore = defineStore("contactmoment", {
   state: () => {
+    const huidigeVraag = initVraag();
     return {
       contactmomentLoopt: false,
-      zaken: <ContactmomentZaak[]>[],
-      klanten: [],
-      notitie: "",
-      contactverzoek: undefined,
-      nieuweKlant: undefined,
-      startdatum: "",
+      vragen: [huidigeVraag],
+      huidigeVraag,
     } as ContactmomentState;
   },
   getters: {
-    klant: (state): Klant | undefined =>
-      state.klanten.filter((x) => x.shouldStore).map((x) => x.klant)[0],
+    klant(state): Klant | undefined {
+      return state.huidigeVraag.klanten
+        ?.filter((x) => x.shouldStore)
+        ?.map((x) => x.klant)?.[0];
+    },
+    huidigeVraagIndex: (state) => state.vragen.indexOf(state.huidigeVraag),
   },
   actions: {
     start() {
+      if (this.contactmomentLoopt) return;
       this.contactmomentLoopt = true;
     },
+    startNieuweVraag() {
+      if (!this.contactmomentLoopt) return;
+      const nieuweVraag = initVraag();
+      if (this.huidigeVraag.klanten) {
+        nieuweVraag.klanten = this.huidigeVraag.klanten.map(
+          (klantKoppeling) => ({
+            ...klantKoppeling,
+          })
+        );
+      }
+      this.vragen.push(nieuweVraag);
+      this.huidigeVraag = nieuweVraag;
+    },
     stop() {
       this.$reset();
       // Temporary. When we implement multiple running contactmomenten, each will have it's own state
       resetAllState();
     },
-    addZaak(zaak: Zaak) {
-      const contactmomentZaak = zaak as ContactmomentZaak;
-      const index = this.zaken.findIndex((element) => element.id === zaak.id);
-      if (index === -1) {
-        //als de zaak nog niet gekoppeld was aan het contact moment dan voegen we hem eerst toe
-        this.zaken.push(contactmomentZaak);
-        contactmomentZaak.shouldStore = true;
-      } else {
-        const existingZaak = this.zaken[index];
-        existingZaak.shouldStore = true;
+    addZaak(zaak: Zaak, vraag: Vraag) {
+      const match = vraag.zaken.find((element) => element.zaak.id === zaak.id);
```

Comment:
`match` => `existingZaak` might be more descriptive.

Same goes for `((element))` => `((contactmomentZaak))`